name: Main Kitty workflow
on:
  push:
    branches:
      - kmikhail
jobs:
  backend_tests:
    runs-on: ubuntu-latest
    services:
      mysql:
        image: mysql:8
        env:
          MYSQL_USER: root
          MYSQL_PASSWORD: ''
          MYSQL_DATABASE: db
    steps:
    - name: Check out code
      uses: actions/checkout@v3
    - name: Set up Python 3.9
      uses: actions/setup-python@v4
      with:
        python-version: 3.9
    - name: Install backend dependencies
      run: |
        python -m pip install --upgrade pip 
        pip install flake8==6.0.0 flake8-isort==6.0.0
        pip install -r ./backend/requirements.txt
    - name: Test with flake8
      run: python -m flake8 backend/
    - name: backend tests
      env:
          MYSQL_USER: root
          MYSQL_PASSWORD: ''
          MYSQL_DATABASE: db
          # Сервер БД запущен в Docker, но его порт проброшен на хост
          # Поэтому подключаемся к 127.0.0.1:5432
          DB_HOST: 127.0.0.1
          DB_PORT: 3306
          SECRET_KEY: 'django-insecure-unm(i#@k_slvz1ri_b2f@sl(8!4ho_h5q%a9rf!l*jpb(qar*)'
          DEBUG: True
          ALLOWED_HOSTS: localhost,127.0.0.1
          LANGUAGE_CODE: 'en'
          TIME_ZONE: 'UTC'
          USE_I18N: True
          USE_L10N: True
          USE_TZ: True
          EMAIL_HOST: 'smtp.gmail.com'
          EMAIL_USE_TLS: False
          EMAIL_PORT: 465
          EMAIL_USE_SSL: True
          EMAIL_HOST_USER: 'wildmv@gmail.com'
          EMAIL_HOST_PASSWORD: '7axn8MLMtg'
      run: |
        cd backend/decksapi/
        python manage.py test